state/secrets.go

```diff
@@ -96,16 +97,23 @@ type secretsStore struct {
 	st *State
 }
 
-func (s *secretsStore) secretMetadataDoc(URL *secrets.URL, p *CreateSecretParams) (*secretMetadataDoc, error) {
+func (s *secretsStore) secretMetadataDoc(p *CreateSecretParams) (*secrets.URL, *secretMetadataDoc, error) {
 	id, err := sequenceWithMin(s.st, "secret", 1)
 	if err != nil {
-		return nil, errors.Trace(err)
+		return nil, nil, errors.Trace(err)
+	}
+	interval := p.RotateInterval
+	if interval < 0 {
+		interval = 0
 	}
-	return &secretMetadataDoc{
+	URL := secrets.NewSimpleURL(p.Version, p.Path)
+	URL.ControllerUUID = p.ControllerUUID
```

Comment:
It seems we do this in both facade and state.
The URL here is like the ID, it's identical for each revision of the secret. 
I would prefer to generate the URL from the facade, then always use that one for the whole process.

Reply:
ok